libbeat/publisher/queue/queue.go

```diff
@@ -72,6 +74,8 @@ type Queue interface {
 	Metrics() (Metrics, error)
 }
 
+type QueueFactory func(logger *logp.Logger, ack func(eventCount int)) (Queue, error)
```

Comment:
Small nit: can we get a comment on the QueueFactory type? Other than that, LGTM.